pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -436,17 +427,26 @@ private void handleLowWaterMark(TxnID txnID, long lowWaterMark) {
         }
     }
 
+    private void takeAbortedTxnSnapshot(Position maxReadPosition) {
+        this.snapshotAbortedTxnProcessor.takeAbortedTxnsSnapshot(this.maxReadPosition)
```

Comment:
This is no need to abstract a new method. Only performing `checkIfNoSnapshot` within the `appendBufferToTxn` method is sufficient.

Reply:
Done.